### DISCUSSION TITLE
ADM5 Klipper Mod v00.03-beta

### DISCUSSION THREAD

**Discussion by xblax** [xblax/flashforge_ad5m_klipper_mod#49] [Announcements]
I guess it's time to finally announce that this mod has reached "beta" status and the repository is opened to the public :tada:

Welcome to all that are new to this project! 

And thank you to all that have already contributed to this project during the last weeks! It took a fair amount of work to get to this point but it was also a lot of fun!  Special thank is due to @consp who has tremendously helped with the development, testing and cool features that are now part of this mod.

There is not much to say about the features of this [Release](https://github.com/xblax/flashforge_adm5_klipper_mod/releases/tag/v00.03-beta), since all the information is in the [Readme](https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/README.md), other documentation, discussions and open issues. 

Everyone is welcome to contribute to this open source project and join the [discussions](https://github.com/xblax/flashforge_adm5_klipper_mod/discussions) here on GitHub!

**Comment by johncotten** [comment]
So far so good! Installed and did an initial print. Thank you all for making this happen!

**Comment by DomanRulez** [comment]
It's possible to print directly from Orca Slicer using this connection setting?
![image](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/161459652/a1d7a79a-f8b8-45c3-b98a-969efc63d05c)

**Reply by the-redone** [reply]
This is basic stuff and all explained already by OP and pretty much basic knowledge, don't think this is for you if you're asking these basic questions 

https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/docs/SLICING.md

**Reply by xblax** [reply]
> This is basic stuff and all explained already by OP and pretty much basic knowledge, don't think this is for you if you're asking these basic questions

I don't mind if some easy questions are asked, maybe not everything is immediately obvious to newcomers.
But the Github Discussions have a [Q&A ](https://github.com/xblax/flashforge_adm5_klipper_mod/discussions/categories/q-a)section where these kind of questions would fit best.

**Reply by consp** [reply]
To answer the original question:

Octo/klipper would be correct, as moonraker has octo-compatibility. your-printer-ip:7125 would be the hostname/ip/url as stated in SLICING.md.

Do note the possible problems with the default orcaslicer settings from flashforge in the docs.

**Reply by the-redone** [reply]
TBH coming from groups on facebook etc it's got to the point where people need to be spoon fed, maybe I'm a bit abrupt but sick of seeing people ask questions that have answers in front of them and relying on others to show them. Ends up in a free for all and answering the same questions over and over, will avoid q&a if that's how it's going to go, appreciate all the work you've done and all collaborators too

**Comment by DougJoseph** [comment]
Thank you so much! Will be trying this out ASAP today!

**Comment by xblax** [comment]
Wrote a short post on Reddit to announce the start of public beta for this mod.
https://new.reddit.com/r/FlashForge/comments/1bcvwpj/custom_klipper_mod_for_flashforge_adventurer_5m/

**Comment by DougJoseph** [comment]
Thank you SO much for making this for all of us in this community!
 
I would really like try the KlipperScreen approach, but I don't yet know how to install that variant. I could not locate a prebuilt file for that variant. I downloaded the source code as a Zip file, and opened the folder in VSCode, but I don't how to build it. 

Please bear in mind I'm brand new to Klipper. I would sure appreciate any kind advice. 

Thank you again for being so awesome!

**Reply by KaruroChori** [reply]
https://github.com/xblax/flashforge_adm5_klipper_mod/releases/tag/v00.03-beta
Third file in there

**Reply by DougJoseph** [reply]
Wow, thanks. Don't know how I missed it!

**Reply by consp** [reply]
> VSCode, but I don't how to build it

I doubt that will work, you definitly need a cross compilation environment which probable means linux is a requirement (or wsl). Building is a bit of a stretch at the moment due to buildroot. Ease of build (plus good docs) is an improvement yet to happen.

**Reply by DougJoseph** [reply]
Re. "I doubt that will work" — I definitely don't want to build myself in VS Code. I just had somehow overlooked the already built file that's provided! @KaruroChori was kind enough to point it out for me. I now have it installed and working! Thanks!

**Comment by DougJoseph** [comment]
I got both the normal variant and the Klipperscreen variant working! Thank you! 

![IMG_7404](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/ff3940c1-d2fe-4774-83c1-ed55aedcbb69)

The only struggle I've had so far is that I could not get connected to wifi even after trying all the options shown in the wifi docs page, with the exception of the final option by use of Klipperscreen. On the klipper_mod method, I'm pretty sure I did it right, but it would not connect (i.e. after reboot). On the SSH method, the commands seemed to work fine, but after reboot no wifi. 

Thanks again for making this!

**Reply by consp** [reply]
> but after reboot no wifi.

I've had this once but could find no a reason for it. Might be worth a look into.

**Comment by DougJoseph** [comment]
Hello again! I'm not a novice at either code or 3D printing, however, I am a novice regarding Klipper, and barely know enough about GCode to be dangerous. I am trying to alter the Flashforge AD5M print profile in **OrcaSlicer** to be suitable for this Klipper mod. I have read and reread this portion in the docs, but don't understand what to do on the last part:

> • GCODE output style should be klipper
> • Do not use the default FlashForge OrcaSlicer settings, they do not contain proper start/end sequences.
> • Edit the printer profile to use the [START_PRINT](https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/printer_configs/macros.cfg#L10) and [END_PRINT](https://github.com/xblax/flashforge_adm5_klipper_mod/blob/master/printer_configs/macros.cfg#L46) as described in macros.cfg as start/end macros.
> Example:
> START_PRINT BED_TEMP={first_layer_bed_temperature[0]} EXTRUDER_TEMP={first_layer_temperature[0]}
> Variables may differ depending on the slicer used.

I am basing this on Flashforge's profile provided for printing the DIY enclosure.

For the first portion, the profile seems to already specify "Klipper" as the GCode Flavor:

<img width="1728" alt="Screenshot 2024-03-12 at 4 31 42 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/559b956c-d848-41dd-83f7-825cbce8857c">

<img width="937" alt="Screenshot 2024-03-12 at 4 32 01 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/457eaf25-d693-4716-a6f4-316eead2a0b7">

**For the last portion, am I supposed to remove all the GCodes and etc that are in the Start and End...** 

<img width="934" alt="Screenshot 2024-03-12 at 4 32 13 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/1eab2643-8fda-459b-954d-8025a53bbf2e">

**...and replace as shown like this?**

<img width="935" alt="Screenshot 2024-03-12 at 4 39 44 PM" src="https://github.com/xblax/flashforge_adm5_klipper_mod/assets/8921397/7db2c4a4-3edb-41ca-aa4e-e56359d15803">

**If not, then would someone be kind enough to explain if incorrect, and/or what else is needed?**

**Thank you so much!**

**Reply by consp** [reply]
NOTE:

> Variables may differ depending on the slicer used.

The start_print mentioned is an example (apperantly it works :)). A tutorial on start codes [here](https://ellis3dp.com/Print-Tuning-Guide/articles/passing_slicer_variables.html) or [here](https://docs.vorondesign.com/community/howto/EricZimmerman/SlicerAndPrintStart.html).

You have to figure it out on your own what variables you exactly need to use. You can probably get them from the original code, but the original start gcode does a M104 which won't wait for the extruder to heat up and will stop before the print starts.

gcode flavor is correct, you don't need to change that.

**Reply by Tiwatz** [reply]
Yes that START_PRINT works fine with Orca slicer.

**Reply by DougJoseph** [reply]
> You have to figure it out on your own what variables you exactly need to use. You can probably get them from the original code, but the original start gcode does a M104 which won't wait for the extruder to heat up and will stop before the print starts.

> Yes that START_PRINT works fine with Orca slicer.

Well, I tried a test print with that "example" code, and something is not right. It tried to heat the nozzle to 255 instead of 220, and it started pressing the nozzle down into the bed really hard. Ugh.

**Reply by xblax** [reply]
Can you provide G-Code that you sliced? Or you can just open it in an Editor and see if the START_PRINT looks as it should. But it would be great if you open a thread in the [Q&A](https://github.com/xblax/flashforge_adm5_klipper_mod/discussions/categories/q-a) section about slicer configuration help.

There that discussion is better suited than here in the release announcement thread.

**Reply by Tiwatz** [reply]
![1](https://github.com/xblax/flashforge_adm5_klipper_mod/assets/162510297/7db448ab-8502-4ef9-b3da-39631a17d400)

This is mine, tested multiple prints today. Probably something else that is wrong for you.

**Reply by DougJoseph** [reply]
OK, seeking to move this over to Q&A here: https://github.com/xblax/flashforge_adm5_klipper_mod/discussions/57

**Reply by DougJoseph** [reply]
Over on the other thread, I managed to solve the "params" issue. Circling back for convenience of readers here.

https://github.com/xblax/flashforge_adm5_klipper_mod/discussions/57#discussioncomment-8765727

**Comment by babyshampoo112** [comment]
i downloaded the klipper screen version, its very good but I connected it to the wifi and whenever i try accessing moonraker or fluidd it says site cant be reached

**Reply by consp** [reply]
Are you sure you are connecting to the correct IP? This sounds more like a network issue to me. There are some issues with klipperscreen not reporting the correct wifi ip though, i.e. see  #39

**Reply by DougJoseph** [reply]
I can confirm that connecting via Moonraker and Fluidd does work once the printer has actually connected to Wifi. My printer did not connect via wifi until I used the Klipperscreen interface on the LCD to input the Wifi login details. All the previous options in the list of how to connect, did not work for me for whatever reason. But once I got connected, the remote interfaces worked.

**Comment by webbbn** [comment]
I've been holding off installing this until I confirmed that my Voron was back up an running after some upgrades, and I just installed the 00.04 release. So far it's working great!

I also hoped to  make sure that Flashforge knows that I only bought this printer because this printer ran klipper, I saw their (maybe halfhearted?) release of the klipper firmware, and saw this repo.

Just finished my first test print after upgrading, and it looks perfect. Great work!

**Reply by consp** [reply]
> maybe halfhearted

Look mode like malicious compliance. 
It's also not possible to build klipper since the analogue parts of the chip won't work. It's therefore impossible to use the repo to build a klipper version for the MCU. It's likely they made changes and did not include them.

But at least it's modable. Really hope someone can get the NS chip SDK somewhere and build for it, would enable everyone to build newer klipper versions.
